src/app/csv-collection/csv-collection.service.ts

```diff
@@ -58,18 +59,44 @@ export class CsvCollectionService {
       }));
   }
 
+  createCsvCollection(csvCollection: CsvCollection): Observable<CsvCollection> {
+    return this.http.post<CsvCollection>(this.csvCollectionUrl, csvCollection);
+  }
+
   getJob(jobUrl: string): Observable<Job> {
     return this.http.get<Job>(jobUrl);
   }
 
-  uploadFile(csvCollection: CsvCollection): Observable<CsvCollection> {
-    const formData = new FormData();
-    for (const file of csvCollection.files) {
-      formData.append('files', file, file.name);
+  getCsv(csvCollection: CsvCollection, params): Observable<PaginatedCsv> {
```

Comment:
`getCsvs` or `getCsvFiles` would be better names here to emphasize that the method is returning an array of CSVs, not a single one